### PR TITLE
CI: add Kernel compile tests

### PR DESCRIPTION
**Pull request by aparcar**
Add Github Actions yaml script to build test kernel PR changes for
each target.

Signed-off-by: Paul Spooren <mail@aparcar.org>
[ add commit description ]
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

### CODE REVIEW

**Comment by Ansuel** [comment]
@aparcar was thinking... why not create a container instead of running debian and installing the package every time? It should speedup the execution

**Comment by aparcar** [comment]
Problem is the broken external toolchain. That should be fixed first.

**Comment by Ansuel** [comment]
@aparcar I just notice that on other action we use `gosu`... here I use another solution as i wasn't aware of that... Don't know what is better.... readability or the contraption to have steps that run on the user directly...

Anyway I tested this on my local repo and in theory it should complete correctly.

I pushed the required changes for external toolchain to work.

**Comment by Ansuel** [comment]
Mh 30 min each to test a kernel in not that bad... guess this needs some tweaks and it's ready...

would be very useful for kernel bump

File: .github/workflows/kernel.yml [hunk 1]
```diff
@@ -95,45 +67,51 @@ jobs:
           echo "SUBTARGET=$SUBTARGET" >> "$GITHUB_ENV"
 
       - name: Update & Install feeds
+        shell: su buildbot -c "sh -e {0}"
+        working-directory: openwrt
```
Comment (aparcar) [inline .github/workflows/kernel.yml:71]: smooth!

Comment (Ansuel) [reply .github/workflows/kernel.yml:71]: Yep no idea if we should use gosudo or this to change the entire shell...imho this is more safe

**Comment by aparcar** [comment]
I think this is great! Could you please cherry pick this into master (except the last commit)?

**Comment by Ansuel** [comment]
Yes sure I will do this in the night as I have to rebase this with the changes for the external toolchain.

**Comment by Ansuel** [comment]
Merged in master with:
8a77adb048 CI: run inside the buildbot docker container
c17c931a90 CI: add Kernel compile tests

**Pull request closed without merging**